### PR TITLE
requstWillBeSent not called for certain requests

### PR DESCRIPTION
**Pull request by durackpl**
Hi Everyone,

I came across this behavior: requstWillBeSent is not called and request object is not created for certain request. 

It seems that this issue is addressed already in _onLoadingFinished/onLoadingFailed function in NetworkManager.js.  However, a similar issue in _onRequestWillBeSent was not addressed.

Here a few lines patch fixing this.

Cheers,
D.

### CODE REVIEW

**Comment by durackpl** [comment]
I've signed it

File: lib/NetworkManager.js [hunk 1]
```diff
@@ -203,6 +203,14 @@ class NetworkManager extends EventEmitter {
     }
     if (event.redirectResponse) {
       const request = this._requestIdToRequest.get(event.requestId);
+
+      // Added (copy similar lines from below): For certain requestIds
+      // we never receive requestWillBeSent event.  @see
+      // https://crbug.com/750469
+      if (!request) {
```
Comment (aslushnikov) [inline lib/NetworkManager.js:210]: When and why does this happen? I tried a few scenarios, but neither worked for me. Network is complicated, I'm very interested in a test for this.

Comment (aslushnikov) [reply lib/NetworkManager.js:210]: For the record: one of my test attempts https://github.com/aslushnikov/puppeteer/commit/3146445bfd2deb2dabd9b3433e243af08dd33a0e

Comment (durackpl) [reply lib/NetworkManager.js:210]: The basic scenario when it happens: (1) click a link which opens a popup; (2) catch popup with 'targetcreated'; (3) click a link on popup; (4) another popup comes up and is caught again through 'targetcreated'; (5) while this second popup is getting loaded, a redirect is received/followed and that is when it happens.

I cannot disclose/inspect internals of the actual webapp where this is observed and I failed to reproduce similar behaviour with simple examples.

Feel free to guide me and suggest further tests I can try on the actual webapp to identify the problem closer.

Comment (aslushnikov) [reply lib/NetworkManager.js:210]: @durackpl do you have any interstitial page in the second pop-up?

Comment (durackpl) [reply lib/NetworkManager.js:210]: As a matter of fact, the first popup seems "interstitial".  It is the first time I see this term "interstitial", and according to wikipedia, the first popup fits "interstitial" definition :)

**Comment by aslushnikov** [comment]
All of a sudden, this has been solved in https://github.com/GoogleChrome/puppeteer/pull/1420.

Thank you for the PR, though!

**Pull request closed without merging**